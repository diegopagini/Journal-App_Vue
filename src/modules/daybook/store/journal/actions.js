/** @format */
import journalApi from '@/api/journalApi';

export const loadEntries = async (/*{ commit }*/) => {
	const { data } = await journalApi.get('/entries.json');
	console.log(data);
};

export const updateEntry = async (/* { commit }*/) => {};

export const createEntry = async (/* { commit }*/) => {};
