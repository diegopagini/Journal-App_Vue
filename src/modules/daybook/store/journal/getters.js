/** @format */

export const getEntriesByTerm =
	(state) =>
	(term = '') => {
		if (term.length === 0) return state.entries;

		return state.entries.filter((entry) => entry.text.toLowerCase().includes(term.toLowerCase()));
	};

export const getEntryById = (/* state */) => {};
