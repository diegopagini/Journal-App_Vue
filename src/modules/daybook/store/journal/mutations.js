/** @format */

export const setEntries = (state, entries) => {
	state.entries = [...state.entries, ...entries];
	state.isLoading = false;
};

export const updateEntry = (/*state*/) => {};

export const createEntry = (/*state*/) => {};
