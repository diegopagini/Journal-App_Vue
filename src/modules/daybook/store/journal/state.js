/** @format */

export default () => ({
	isLoading: false,
	entries: [
		{
			id: '1',
			date: new Date().toDateString(),
			text: 'Hola mundo',
			picture: null,
		},
		{
			id: '2',
			date: new Date().toDateString(),
			text: 'Cillum sunt minim deserunt eu cupidatat deserunt fugiat culpa tempor sunt laborum reprehenderit aliqua sint. Do in ea reprehenderit do. Excepteur anim aliqua sunt deserunt et excepteur voluptate cupidatat labore labore excepteur. Minim commodo deserunt irure sunt consectetur irure fugiat tempor labore culpa ut sunt et. Enim non in velit quis culpa deserunt esse nostrud incididunt et qui consequat. Reprehenderit esse cillum eiusmod sint consequat non eu.',
			picture: null,
		},
		{
			id: '3',
			date: new Date().toDateString(),
			text: 'Cillum sunt minim deserunt eu cupidatat deserunt fugiat culpa tempor sunt laborum reprehenderit aliqua sint. Do in ea reprehenderit do. Excepteur anim aliqua sunt deserunt et excepteur voluptate cupidatat labore labore excepteur. Minim commodo deserunt irure sunt consectetur irure fugiat tempor labore culpa ut sunt et. Enim non in velit quis culpa deserunt esse nostrud incididunt et qui consequat. Reprehenderit esse cillum eiusmod sint consequat non eu.',
			picture: null,
		},
		{
			id: '4',
			date: new Date().toDateString(),
			text: 'Cillum sunt minim deserunt eu cupidatat deserunt fugiat culpa tempor sunt laborum reprehenderit aliqua sint. Do in ea reprehenderit do. Excepteur anim aliqua sunt deserunt et excepteur voluptate cupidatat labore labore excepteur. Minim commodo deserunt irure sunt consectetur irure fugiat tempor labore culpa ut sunt et. Enim non in velit quis culpa deserunt esse nostrud incididunt et qui consequat. Reprehenderit esse cillum eiusmod sint consequat non eu.',
			picture: null,
		},
		{
			id: '5',
			date: new Date().toDateString(),
			text: 'Cillum sunt minim deserunt eu cupidatat deserunt fugiat culpa tempor sunt laborum reprehenderit aliqua sint. Do in ea reprehenderit do. Excepteur anim aliqua sunt deserunt et excepteur voluptate cupidatat labore labore excepteur. Minim commodo deserunt irure sunt consectetur irure fugiat tempor labore culpa ut sunt et. Enim non in velit quis culpa deserunt esse nostrud incididunt et qui consequat. Reprehenderit esse cillum eiusmod sint consequat non eu.',
			picture: null,
		},
	],
});
