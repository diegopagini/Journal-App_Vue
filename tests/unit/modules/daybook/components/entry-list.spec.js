/** @format */
import EntryList from '@/modules/daybook/components/entry-list.vue';
import journal from '@/modules/daybook/store/journal';
import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';

import { journalState } from '../../../mock-data/test-journal-state';

const createVuexStore = (initialState) =>
	createStore({
		modules: {
			journal: {
				...journal,
				state: { ...initialState },
			},
		},
	});

describe('Entry List component', () => {
	let wrapper;

	const router = {
		push: jest.fn(),
	};

	const store = createVuexStore(journalState);

	beforeEach(() => {
		jest.clearAllMocks();

		wrapper = shallowMount(EntryList, {
			global: {
				mocks: {
					$router: router,
				},
				plugins: [store],
			},
		});
	});

	it('Should call getEntriesByTermn and show all the entries', () => {
		expect(wrapper.findAll('entry-stub').length).toEqual(4);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should call the getEntriesByTermn and filter the entries', async () => {
		const input = wrapper.find('input');
		await input.setValue('Prueba');
		expect(wrapper.findAll('entry-stub').length).toEqual(1);
	});

	it('the "new" button should redirect to /new', () => {
		wrapper.find('button').trigger('click');
		expect(router.push).toHaveBeenCalledWith({ name: 'entry', params: { id: 'new' } });
	});
});
