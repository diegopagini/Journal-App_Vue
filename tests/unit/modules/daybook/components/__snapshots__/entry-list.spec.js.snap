// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry List component Should call getEntriesByTermn and show all the entries 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input class="form-control" placeholder="Buscar entrada" type="text"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva Entrada </button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
